File: bro-agent-api/src/main/java/com/ericsson/adp/mgmt/bro/api/grpc/BackupFileMessageBuilder.java
/**------------------------------------------------------------------------------
 *******************************************************************************
 * COPYRIGHT Ericsson 2019
 *
 * The copyright to the computer program(s) herein is the property of
 * Ericsson Inc. The programs may be used and/or copied only with written
 * permission from Ericsson Inc. or in accordance with the terms and
 * conditions stipulated in the agreement/contract under which the
 * program(s) have been supplied.
 *******************************************************************************
 *------------------------------------------------------------------------------*/
package com.ericsson.adp.mgmt.bro.api.grpc;

import java.util.function.Consumer;

import com.ericsson.adp.mgmt.data.BackupData;
import com.ericsson.adp.mgmt.data.BackupFileChunk;
import com.ericsson.adp.mgmt.data.BackupFileChunk.Builder;
import com.ericsson.adp.mgmt.data.DataMessageType;
import com.google.protobuf.ByteString;

/**
 * Builds messages to send backup file through Data Channel.
 */
public class BackupFileMessageBuilder implements BackupMessageBuilder {

    @Override
    public BackupData getFileNameMessage(final String fileName) {
        return generateMessage(backupFileChunkBuilder -> backupFileChunkBuilder.setFileName(fileName));
    }

    @Override
    public BackupData getDataMessage(final ByteString data) {
        return generateMessage(backupFileChunkBuilder -> backupFileChunkBuilder.setContent(data));
    }

    @Override
    public BackupData getChecksumMessage(final String checksum) {
        return generateMessage(backupFileChunkBuilder -> backupFileChunkBuilder.setChecksum(checksum));
    }

    private BackupData generateMessage(final Consumer<Builder> setMessageInformationFunction) {
        final Builder messageBuilder = BackupFileChunk.newBuilder();
        setMessageInformationFunction.accept(messageBuilder);
        return wrapMessage(messageBuilder.build());
    }

    private BackupData wrapMessage(final BackupFileChunk chunk) {
        return BackupData
                .newBuilder()
                .setBackupFileChunk(chunk)
                .setDataMessageType(DataMessageType.BACKUP_FILE)
                .build();
    }

}

File: bro-agent-api/src/main/python/bro_agent/grpc/BackupFileMessageBuilder.py
import bro_agent.generated.INT_BR_ORCH_DATA_pb2 as INT_BR_ORCH_DATA
from bro_agent.grpc.BackupMessageBuilder import BackupMessageBuilder


class BackupFileMessageBuilder(BackupMessageBuilder):
    """ To build the backup file message"""

    @staticmethod
    def get_file_name_message(file_name):
        """ get the file name """
        backup_file_chunk = INT_BR_ORCH_DATA.BackupFileChunk(fileName=file_name)
        backup_data = INT_BR_ORCH_DATA.BackupData(dataMessageType=INT_BR_ORCH_DATA.DataMessageType.BACKUP_FILE,
                                                  backupFileChunk=backup_file_chunk)
        return backup_data

    @staticmethod
    def get_data_message(data):
        """ get data message """
        backup_file_chunk = INT_BR_ORCH_DATA.BackupFileChunk(content=data)
        backup_data = INT_BR_ORCH_DATA.BackupData(dataMessageType=INT_BR_ORCH_DATA.DataMessageType.BACKUP_FILE,
                                                  backupFileChunk=backup_file_chunk)
        return backup_data

    @staticmethod
    def get_checksum_message(checksum):
        """"get checksum message """
        backup_file_chunk = INT_BR_ORCH_DATA.BackupFileChunk(checksum=checksum)
        backup_data = INT_BR_ORCH_DATA.BackupData(dataMessageType=INT_BR_ORCH_DATA.DataMessageType.BACKUP_FILE,
                                                  backupFileChunk=backup_file_chunk)
        return backup_data
